Sources/XcodeGraph/Models/Target.swift

```diff
@@ -217,6 +218,27 @@ public struct Target: Equatable, Hashable, Comparable, Codable, Sendable {
         }
     }
 
+    /// This function validates the sources against other target metadata returning which sources from the list
+    /// are valid and invalid.
+    /// - Returns: A list of valid and invalid sources.
+    public func validatedSources() -> (valid: [SourceFile], invalid: [SourceFile]) {
```

Comment:
wdyt about moving this to `tuist/tuist` instead? I don't see this as high value having in `XcodeGraph` and any convenience function we end up having here means we need to keep backwards compatible.

Reply:
Good idea. I removed it from here and marked the existing `supportsSources` as deprecated.